### PR TITLE
[DA-3566] Populate Death Table from HealthPro Deceased Data

### PR DESCRIPTION
**Pull request by brianmkimmel**
## Description of changes/additions

For CDR v8 deceased data captured in HealthPro needs to be sent through the ETL for CDR use in the research workbench.

Updates the ETL process to populate the cdm.death table from approved rdr.deceased_report records.

## Tests
- [x] unit tests

[DA-3566]: https://precisionmedicineinitiative.atlassian.net/browse/DA-3566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rdr_service/tools/tool_libs/curation.py [hunk 1]
```diff
@@ -702,6 +703,30 @@ def _select_participant_ids(self, session, origin:str, cutoff_date:datetime=None
...
+            Death.cause_concept_id: literal(None),
+            Death.cause_source_value: literal(None),
+            Death.cause_source_concept_id: literal(None),
+            Death.src_id: literal("healthpro")
+        }
+        deceased_select = session.query(*column_map.values()).select_from(
+            DeceasedReport
+        ).join(
+         Person,
+         DeceasedReport.participantId == Person.person_id
```
Comment (kenny-skaggs) [inline rdr_service/tools/tool_libs/curation.py]: :nit Please clean up the indentation here

**Review by j-kanuch — APPROVED** [review]

File: rdr_service/tools/tool_libs/curation.py [hunk 2]
```diff
@@ -702,6 +703,30 @@ def _select_participant_ids(self, session, origin:str, cutoff_date:datetime=None
...
 
+    def _populate_death_table(self, session: sqlalchemy.orm.session):
+        # Populates death table from deceased_report
+        self._set_rdr_model_schema([DeceasedReport])
+        column_map = {
+            Death.id: literal("0"),
+            Death.person_id: DeceasedReport.participantId,
+            Death.death_date: DeceasedReport.dateOfDeath,
+            Death.death_datetime: DeceasedReport.dateOfDeath.label('date_of_death_datetime'),
+            Death.death_type_concept_id: literal("32809"),
```
Comment (kenny-skaggs) [inline rdr_service/tools/tool_libs/curation.py]: Please add a comment to indicate what the 32809 concept is.

File: rdr_service/tools/tool_libs/curation.py [hunk 3]
```diff
@@ -702,6 +703,30 @@ def _select_participant_ids(self, session, origin:str, cutoff_date:datetime=None
         result = query.order_by(Participant.participantId).all()
         self.pid_list = [pid[0] for pid in result]
 
+    def _populate_death_table(self, session: sqlalchemy.orm.session):
+        # Populates death table from deceased_report
+        self._set_rdr_model_schema([DeceasedReport])
+        column_map = {
+            Death.id: literal("0"),
```
Comment (kenny-skaggs) [inline rdr_service/tools/tool_libs/curation.py]: The `id` column is a primary key and would throw an error if any more than one record is inserted. Should the `id` column be auto generated instead?

Comment (brianmkimmel) [reply rdr_service/tools/tool_libs/curation.py]: id is an auto-increment column, MySQL picks the next sequence number when 0 is inserted. I've added a comment to note that.

Comment (kenny-skaggs) [reply rdr_service/tools/tool_libs/curation.py]: Thanks! I wasn't aware of that.

**Review by kenny-skaggs — APPROVED** [review]

**Pull request merged**